src/mpid/ch4/netmod/ucx/subconfigure.m4

```diff
@@ -26,6 +26,21 @@ AM_COND_IF([BUILD_CH4_NETMOD_UCX],[
     ucxlib=""
     AC_SUBST([ucxlib])
 
+    if test "$pac_have_ucx" = "yes" -a "$with_ucx" != "embedded"; then
+        AC_MSG_CHECKING([if ucx meets minimum version requirement])
+        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([#include <ucp/api/ucp.h>], [
+                           #if UCP_VERSION(UCP_API_MAJOR, UCP_API_MINOR) < UCP_VERSION(1, 7)
+                           #error
+                           #endif
+                           return 0;])],[],[pac_have_ucx=no])
+        AC_MSG_RESULT([$pac_have_ucx])
+
+        # if an old ucx was specified by the user, throw an error
+        if test "$pac_have_ucx" = "no" -a -n "$with_ucx" -a "$with_ucx" != "yes" ; then
```

Comment:
Maybe we need to move this check into the "non-embedded" branch later.

I think we always need an error message in this case (`$pack_have_ucx = no`). For better error messages, we may have to run two tests, one just probe ucx header, the second one check versions and we need tell the user what is the required minimum version.

Reply:
OK, will do.